tests/integration-tests/utils.py

```diff
@@ -127,6 +128,51 @@ def random_alphanumeric(size=16):
     return "".join(random.choice(string.ascii_lowercase + string.digits) for _ in range(size))
 
 
+def retrieve_metric_data(unique_name, cluster_name, metric_name, period_length_sec, collection_time_min):
```

Comment:
Some of the parameters here are lists but the. name gives the impression that they're strings.
`unique_name` is meant to be a list of `Metric Ids`, right?
`metric_name` is meant to be a list of Metric Names, right?

Also, this might be out of scope for now but this function can be made to be generic e.g. instead of `cluster_name`, you can have `namespace` and let the caller specify the namespace. Other arguments such as the stat, aws_region can also be made into keyword arguments with defaults.

Reply:
No ID here is just a unique string that distinguishes the query from others, it doesn't have to reflect the metric at all. Its kinda the name of the query. The metric name is used later in the get_metric_data call. 

The addition of namespace I think is a great idea, unfortunately since the feature hasn't been implemented by the cloud-formation team as of yet, I cant add that just now.  The idea to add other arguments such as the stat, and aws_region is a good idea and will be implemented